Framework/API/test/WorkspaceListPropertyTest.h

```diff
@@ -0,0 +1,293 @@
+
+    virtual void init() {
+      declareProperty(std::make_unique<WorkspaceListProperty<MatrixWorkspace>>(
+          "MyProperty", std::vector<MatrixWorkspace_sptr>(0)));
+    }
+
+    virtual void exec() {
+      std::vector<MatrixWorkspace_sptr> val = getProperty("MyProperty");
+    }
+
+    virtual ~MyAlgorithm2() {}
+  };
+
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static WorkspaceListPropertyTest *createSuite() {
+    return new WorkspaceListPropertyTest();
+  }
+  static void destroySuite(WorkspaceListPropertyTest *suite) { delete suite; }
+
+  //------------------------------------------------------------------------------
+  // Functional Testing
+  //------------------------------------------------------------------------------
+
+  void test_constructor() {
+    auto list = createWorkspaceList();
+
+    TS_ASSERT_THROWS_NOTHING(std::make_unique<WorkspaceListProperty<Workspace>>(
+        WorkspaceListProperty<Workspace>("MyWorkspaceProperty", list,
+                                         Direction::Input,
+                                         PropertyMode::Mandatory)));
+  }
+
+  void test_contruct_as_optional() {
+    auto list = std::vector<Workspace_sptr>();
+
+    TS_ASSERT_THROWS_NOTHING(std::make_unique<WorkspaceListProperty<Workspace>>(
+        WorkspaceListProperty<Workspace>("MyWorkspaceProperty", list,
+                                         Direction::Input,
+                                         PropertyMode::Optional)));
+  }
+
+  void test_copy_construction() {
+    auto list = createWorkspaceList();
+
+    WorkspaceListProperty<Workspace> a("PropA", list, Direction::Input,
+                                       PropertyMode::Optional);
+    WorkspaceListProperty<Workspace> b(a);
+
+    TS_ASSERT_EQUALS(list, a.list());
+    TS_ASSERT_EQUALS(a.list(), b.list());
+    TS_ASSERT_EQUALS(a.isOptional(), b.isOptional());
+  }
+
+  void test_assignment() {
+    auto list = createWorkspaceList();
+    WorkspaceListProperty<Workspace> propA("PropA", list, Direction::Input,
+                                           PropertyMode::Mandatory);
+    WorkspaceListProperty<Workspace> propB(
+        "PropB", std::vector<Workspace_sptr>(), Direction::Input,
+        PropertyMode::Optional);
+    propB = propA;
+    TS_ASSERT_EQUALS(list, propA.list());
+    TS_ASSERT_EQUALS(propA.list(), propB.list());
+    TS_ASSERT_EQUALS(propA.isOptional(), propB.isOptional());
+  }
+
+  void test_clone() {
+    auto list = createWorkspaceList();
+    WorkspaceListProperty<Workspace> *propOne =
+        new WorkspaceListProperty<Workspace>("PropA", list, Direction::Input,
+                                             PropertyMode::Optional);
+    WorkspaceListProperty<Workspace> *propTwo = propOne->clone();
+    TS_ASSERT_DIFFERS(propOne, propTwo);
+    delete propOne;
```

Comment:
Minor detail. There's no reason to create `propOne` on the heap. You can just create it on the stack and then you don't need the `delete`